src/components/LanguageSelection/LanguageSelection.js

```diff
@@ -0,0 +1,104 @@
+import React, { useContext, useState } from "react";
+import { ConfigContext } from "Context";
+import { useAsync } from "react-use";
+import Loader from "components/Loader";
+import { FormattedMessage, useIntl } from "react-intl";
+import { v4 as uuidv4 } from "uuid";
+import {
+  ButtonGroup,
+  Heading,
+  LinkButton,
+} from "@utrecht/component-library-react";
+import { get, put } from "../../api";
+import PropTypes from "prop-types";
+
+const propTypes = PropTypes.shape({
+  heading: PropTypes.string,
+  headingLevel: PropTypes.number,
+  onLanguageChange: PropTypes.func,
+});
+
+const LanguageSelection = (props) => {
+  // Hook uses
+  const config = useContext(ConfigContext);
+  const { locale } = useIntl();
+  const [current, setCurrent] = useState(locale);
+  const getAvailableLanguages = async () => {
+    const result = await get(`${config.baseUrl}i18n/info`);
+    setCurrent(result.current); // state change if current changed
+    return result.languages.map(({ code, name }) => ({
+      lang: code,
+      textContent: code.toUpperCase(),
+      label: name,
+      current: code === result.current,
+    }));
+  };
+  const {
+    loading,
+    value: items,
+    error,
+  } = useAsync(
+    getAvailableLanguages,
+    [config.baseUrl] // actually current too?!
+  );
+  if (error) {
+    throw error; // bubble up to boundary
+  }
+  if (loading) {
+    return <Loader modifiers={["small"]} />;
+  }
+
+  // unpack props
+  const { heading, headingLevel, onLanguageChange } = props;
+  const headingId = uuidv4(); // XXX: useId() from 'react'
+
+  const changeLanguage = (lang_code) => async () => {
```

Comment:
```suggestion
  const onChangeLanguage = (lang_code) => async () => {
      event.preventDefault();
```
1. event callbacks should be prefixed with `on` (convention)

I initially also had the `event` argument added to prevent accidental navigations/submissions, but that shouldn't be a problem when you use `<button type="button">` elements.